### ISSUE TITLE
openresty gzip response do not work

### ISSUE THREAD

**Issue by hengqujushi** [openresty/lua-nginx-module#984]
I am using Nginx as reverse proxy to Openresty, because the response is large(almost 20kb ),I want to compress the response.

My Nginx config (about gzip) is : 

proxy_set_header Accept-Encoding 'gzip';
gzip  on;
gzip_types  text/plain text/css application/x-javascript text/xml application/xml application/xml+rss text/javascript application/json;
gzip_proxied any;

My Openresty config (about gzip) is : 

gzip  on;
gzip_types text/plain text/css application/x-javascript text/xml application/xml application/xml+rss text/javascript application/json;

when I visit through ip:port(openresty), I found it's compressed:
HTTP/1.1 200 OK
Server: openresty/1.11.2.2
Date: Mon, 13 Feb 2017 03:26:14 GMT
Content-Type: application/json; charset=utf-8
Transfer-Encoding: chunked
Connection: keep-alive
Access-Control-Allow-Origin: *
Content-Encoding: gzip

but when I visit use domain which through nginx-->openresty， I found it's not compressed:
HTTP/1.1 200 OK
Date: Mon, 13 Feb 2017 03:23:26 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 498
Connection: keep-alive
Access-Control-Allow-Origin: *

**Comment by agentzh** [comment]
@hengqujushi It's apparently your nginx side's configuration problem since openresty is working in isolation as you've found out.

**Comment by agentzh** [comment]
I don't know why you are reporting this to openresty's lua-nginx-module.

**Comment by hengqujushi** [comment]
@agentzh I am so sorry , and with your remind, I fix my problem.

In order to help someone who run into the same problem, I still want to give my solution below , and I hope this don't give you more trouble.

I fixed it by adding "gzip_http_version 1.0;" in my nginx configuration, after I found that when nginx serve as reverse proxy, it use http 1.0 to transport data to upstream.

Beyond that, mime.types should include "application/json json;".
